src/core/grid_scan_matcher.h

```diff
@@ -8,45 +14,99 @@
 #include "state_data.h"
 #include "sensor_data.h"
 
+/**
+ * \brief Interface of observer on Scan Matcher
+ */
 class GridScanMatcherObserver {
 public:
+  /**
+   * Virtual function. Starts matching.
```

Comment:
Consider: "A callback that is invoked on scan matching start". Apply the same to the comments below.